src/test/scala/chiselTests/MigrateCompileOptionsSpec.scala

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chiselTests
+
+import chisel3.stage.ChiselStage
+
+import org.scalatestplus.scalacheck.ScalaCheckDrivenPropertyChecks
+
+class MigrateCompileOptionsSpec extends ChiselFlatSpec with ScalaCheckDrivenPropertyChecks with Utils {
+  import Chisel.{defaultCompileOptions => _, _}
+  import chisel3.RequireSyncReset
+
+  behavior.of("Migrating infer resets")
+
+  val migrateIR = new chisel3.CompileOptions {
+    val connectFieldsMustMatch = false
+    val declaredTypeMustBeUnbound = false
+    val dontTryConnectionsSwapped = false
+    val dontAssumeDirectionality = false
+    val checkSynthesizable = false
+    val explicitInvalidate = false
+    val inferModuleReset = false
+
+    override val migrateInferModuleReset = true
+  }
```

Comment:
If we're going to adopt this new technique of "migration" compile options, I think we should consider some factory methods to construct them from `Strict` and `NotStrict`, I'm thinking something like:

```scala
object CompileOptions {
  def migrationFromStrict(migrateInferModuleReset: Boolean = false): CompileOptions = ...
}
```

Anyway, I don't think it should be part of this PR, but if we are to sort of standardize on this technique I think a little bit of help for users would be warranted, otherwise this is a very power user-y API.